### PR TITLE
refact: enhance personal rank

### PR DESCRIPTION
**Pull request by imbajin**

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PersonalRankAPI.java [hunk 1]
```diff
@@ -105,13 +108,15 @@ public String personalRank(@Context GraphManager manager,
...
+        private double alpha = 0.85;
+        @JsonProperty("max_diff")
+        private double maxDiff = 0.0001;
         @JsonProperty("max_degree")
         private long maxDegree = Long.parseLong(DEFAULT_MAX_DEGREE);
         @JsonProperty("limit")
         private long limit = Long.parseLong(DEFAULT_LIMIT);
         @JsonProperty("max_depth")
-        private int maxDepth;
+        private int maxDepth = 5;
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PersonalRankAPI.java]: define a const

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/PathsTraverser.java [hunk 2]
```diff
@@ -61,19 +62,21 @@ public PathSet paths(Id sourceV, Directions sourceDir,
...
             }
-            traverser.forward(sourceDir);
+            findCycle = traverser.forward(targetV, sourceDir);
 
-            if (--depth < 0 || traverser.reachLimit()) {
+            if (--depth < 0 || findCycle || traverser.reachLimit()) {
                 break;
             }
-            traverser.backward(targetDir);
+            findCycle = traverser.backward(sourceV, targetDir);
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/PathsTraverser.java]: finish traverse immediately if find cycle？

**Review by zhoney — CHANGES_REQUESTED** [review]

**Comment by zhoney** [comment]
rocksdb ci failed. At lease ensure memory and rocksdb ci pass

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/PathsTraverser.java (hunk 2 above)
Comment (imbajin) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/PathsTraverser.java]: split to another pr

**Review by coderzc — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

**Pull request merged**